thinc/backends/ops.py

```diff
@@ -261,31 +278,20 @@ def unflatten(self, X: Floats2d, lengths: Ints1d, pad: int = 0) -> List[Floats2d
         for length in lengths:
             length = int(length)
             if pad >= 1 and length != 0:
-                X = X[pad:]
+                X = cast(ArrayTXd, X[pad:])
             unflat.append(X[:length])
-            X = X[length:]
+            X = cast(ArrayTXd, X[length:])
         if pad >= 1:
-            X = X[pad:]
+            X = cast(ArrayTXd, X[pad:])
         assert len(X) == 0
         assert len(unflat) == len(lengths)
-        return unflat
-
-    @overload
-    def pad(self, seqs: List[Ints2d], round_to=1) -> Ints3d:
-        ...
-
-    @overload  # noqa: F811
-    def pad(self, seqs: List[Floats2d], round_to=1) -> Floats3d:
-        ...
+        return cast(List[ArrayTXd], unflat)
 
-    def pad(  # noqa: F811
-        self, seqs: Union[List[Ints2d], List[Floats2d]], round_to=1
-    ) -> Array3d:
+    def pad(self, seqs: List[ArrayT123d_co], round_to=1) -> ArrayT234d:
```

Comment:
I'm unsure about the expansion to other dimensions here. The comment in the code seems to imply that more work/checks are need to make sure this works. Perhaps it's too much to try and get that included also in this PR? At the very least, there should be unit tests added that check the behaviour of `pad`/`unpad` and we'd need to carefully consider how the layer implementations will be impacted. I suggest to split these changes off into separate PRs.

To keep the same functionality as before, can we instead retype this with `ArrayT2d_s` as well, to support only lists of ints, or lists of floats?

Reply:
Yes, this was basically a mistake and I've now removed it.